src/www/ui_components/server_card.ts

```diff
@@ -0,0 +1,286 @@
+        >
+          [[connectButtonLabel]]
+        </paper-button>
+      </div>
+    </paper-card>
+  `;
+
+  @property({type: Boolean}) disabled: boolean;
+  @property({type: String}) errorMessage: string;
+  @property({type: Boolean}) expanded = false;
+  @property({type: String}) rootPath: string;
+  @property({type: String}) serverAddress: string;
+  @property({type: String}) serverId: string;
+  @property({type: String}) serverName: string;
+  @property({type: String}) state: ServerCardState = ServerCardState.DISCONNECTED;
+
+  // Need to declare localize function passed in from parent, or else
+  // localize() calls within the template won't be updated.
+
+  // @polymer/decorators doesn't support Function constructors...
+  @property({type: Object}) localize: (unlocalizedText: string) => string;
+
+  @computed('state', 'localize')
+  get statusMessage() {
+    if (!this.localize) return '';
+
+    switch (this.state) {
+      case ServerCardState.CONNECTING:
+        return this.localize('connecting-server-state');
+      case ServerCardState.CONNECTED:
+        return this.localize('connected-server-state');
+      case ServerCardState.RECONNECTING:
+        return this.localize('reconnecting-server-state');
+      case ServerCardState.DISCONNECTING:
+        return this.localize('disconnecting-server-state');
+      case ServerCardState.DISCONNECTED:
+      default:
+        return this.localize('disconnected-server-state');
+    }
+  }
+
+  @computed('state', 'localize')
+  get connectButtonLabel() {
+    if (!this.localize) return '';
+
+    switch (this.state) {
+      case ServerCardState.CONNECTING:
+      case ServerCardState.CONNECTED:
+      case ServerCardState.RECONNECTING:
+        return this.localize('disconnect-button-label');
+      case ServerCardState.DISCONNECTING:
+      case ServerCardState.DISCONNECTED:
+      default:
+        return this.localize('connect-button-label');
+    }
+  }
+
+  @computed('state')
+  get connectButtonDisabled() {
+    return (
+        this.disabled || this.state === ServerCardState.CONNECTING ||
+        this.state === ServerCardState.DISCONNECTING);
+  }
+
+  @computed('expanded')
+  get expandedClassName() {
+    return this.expanded ? 'expanded' : '';
+  }
+
+  protected onConnectToggled() {
+    this.state === ServerCardState.DISCONNECTED ? this.fireConnectPressed() :
+                                                  this.fireDisconnectPressed();
+  }
+
+  protected onMenuItemPressed({detail: {selected}}: CustomEvent) {
+    if (selected === 'forget') {
+      this.fireForgetRequest();
+    } else if (selected === 'rename') {
+      this.fireShowServerRename();
+    }
+
+    // This can leave the pressed paper-item in the selected state,
+    // causing it to get selected styling (e.g. font-weight: bold),
+    // so explicitly deselect it:
+    this.async(() => (this.$.menu as HTMLInputElement).select());
+  }
+
+  private fireConnectPressed() {
```

Comment:
These functions only add indirection, hurting readability. I'd inline them instead.

Reply:
Totally, as long as we're consistent one way or the other - before we had a combination of inlining and separate functions